src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -1,23 +1,72 @@
 import _ from 'lodash/fp'
 import { useState } from 'react'
-import { div, h, label, span } from 'react-hyperscript-helpers'
+import { div, h, label, p, span } from 'react-hyperscript-helpers'
 import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common'
+import { warningBoxStyle } from 'src/components/data/data-utils'
+import { icon } from 'src/components/icons'
 import { TextInput } from 'src/components/input'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
-import { bucketBrowserUrl } from 'src/libs/auth'
+import { bucketBrowserUrl, getUser } from 'src/libs/auth'
+import colors from 'src/libs/colors'
 import { reportError } from 'src/libs/error'
+import { isAppDeletable } from 'src/libs/runtime-utils'
 import * as Utils from 'src/libs/utils'
 
 
 const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucketName } }, onDismiss, onSuccess }) => {
   const [deleting, setDeleting] = useState(false)
   const [deleteConfirmation, setDeleteConfirmation] = useState('')
+  const [loadingApps, setLoadingApps] = useState(true)
+  const [apps, setApps] = useState()
+
+  const signal = Utils.useCancellation()
+
+  const loadApps = async workspaceName => {
+    const [currentWorkspaceAppList] = await Promise.all([
+      Ajax(signal).Apps.listWithoutProject({ creator: getUser().email, saturnWorkspaceName: workspaceName })
+    ])
+    setApps(currentWorkspaceAppList)
+    setLoadingApps(false)
+  }
+
+  Utils.useOnMount(() => {
+    loadApps(name)
+  })
+
+  const getDeletableApps = () => _.filter(isAppDeletable, apps)
+
+  const getNonDeletableApps = () => _.filter(app => !isAppDeletable(app), apps)
```

Comment:
This iterates through `apps` multiple times. Consider:
```suggestion
  const [deletableApps, nonDeletableApps] = _.partition(isAppDeletable, apps)
```